src/components/DropZone.tsx

```diff
@@ -27,28 +28,37 @@ export function DropZoneContainer(props: {
   );
 }
 
-export function DropZone(props: DropzoneProps) {
-  const { color = 'black', onDrop } = props;
-  return <DropZoneContent color={color} onDrop={onDrop} isContainer={false} />;
+export function DropZone(props: DropZoneProps) {
+  return <DropZoneContent {...props} />;
 }
 
-function DropZoneContent(props: {
-  children?: JSX.Element;
-  isContainer: boolean;
-  color?: string;
-  onDrop: (files: File[]) => void;
-  onClick?: MouseEventHandler<HTMLDivElement>;
-}) {
-  const { color = 'black', children, onDrop, isContainer, onClick } = props;
+function DropZoneContent(
+  props: DropZoneProps & {
+    children?: JSX.Element;
+    isContainer?: boolean;
+    onClick?: MouseEventHandler<HTMLDivElement>;
+  },
+) {
+  const {
+    color = 'black',
+    children,
+    onDrop,
+    isContainer = false,
+    onClick,
+    fileValidator,
+    maxFiles,
+  } = props;
 
   const handleOnDrop = useCallback(
-    (acceptedFiles) => {
-      onDrop(acceptedFiles);
+    <T extends File>(acceptedFiles: T[]) => {
+      onDrop?.(acceptedFiles);
     },
     [onDrop],
   );
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    validator: fileValidator,
```

Comment:
Could you give an example of fileValidator in the story book ? Not sure how to use it. could we filter out to keep only 'txt' extensions ?

We have many (or should at least) have filtering capabilities in filelist-utils so I'm not sure we need to deal with this here.

Reply:
> Could you give an example of fileValidator in the story book ?

Yes okey no problem